agent/src/flow_generator/protocol_logs/mq/kafka.rs

```diff
@@ -542,14 +550,18 @@ impl KafkaLog {
         }
 
         if trace_id.len() > 0 {
-            if let Some(end_index) = trace_id.find("-") {
-                info.trace_id = decode_base64_to_string(&trace_id[..end_index]);
+            let segs: Vec<&str> = trace_id.split('-').collect();
+            if segs.len() >= 1 {
+                info.trace_id = decode_base64_to_string(segs[0]);
+            }
+            if segs.len() >= 2 {
+                info.span_id = segs[1].to_string();
             }
```

Comment:
避免不必要的`collect`

```suggestion
            let segs = trace_id.split('-');
            if let Some(seg) = segs.next() {
                info.trace_id = decode_base64_to_string(seg);
            }
            if let Some(seg) = segs.next() {
                info.span_id= decode_base64_to_string(seg);
            }
```

后同